fix: change dirty strategy in check form

ckbedwell opened the pull request:
Closes: https://github.com/grafana/synthetic-monitoring-app/issues/1026#event-15826703952

## Problem

When introducing the notify unsaved changes feature, an accidental bug was introduced where after using the test button the form was considered submitted and could no longer be saved.

## Solution

The solution is to change the strategy in assessing whether the form is 'dirty' or not. Instead of utilizing React hook form's `isDirty` value in provided by the `formState`, we are manually checking if the initial values match the current values. It is not the most optimal solution but it does work adequately. 

The desired solution proposed in the issue of not considering the test button a submit button has its own tradeoffs with the way `react-hook-form` works and the current set-up of the Synthetic Monitoring application. Because the `FormLayout` component doesn't expose any of its inner workings (such as moving between steps and the validation invalid / valid triggers provided by `react-hook-form`'s `handleSubmit`) it would require a larger re-work of these components and have the same UX.

The `test` button is likely to evolve when we tackle phase 2 of the check creation flow so this solution is most appropriate right now.

Last note, it is difficult to write tests for the 'test' button because JSDom doesn't expose the submitter element, which we rely upon to know if it is an ad-hoc test or saving the check.

VikaCep commented:
There is still a strange behavior going on. When there are changes, the form correctly enables the Save button. But when clicking save, the message indicating there are unsaved changes is prompted asking to stay/leave the page. It still saves but it's confusing as there's no clear indication if the save is successful.

https://github.com/user-attachments/assets/47b26049-a561-49d8-97c0-722800f6b82a

VikaCep reviewed (changes requested):
https://github.com/grafana/synthetic-monitoring-app/pull/1037#issuecomment-2573515581

ckbedwell commented:
> There is still a strange behavior going on. When there are changes, the form correctly enables the Save button. But when clicking save, the message indicating there are unsaved changes is prompted asking to stay/leave the page. It still saves but it's confusing as there's no clear indication if the save is successful.
> 
>  Screen.Recording.2025-01-06.at.13.51.33.mov

I'm glad you had the common sense to check submitting it (because I didn't) 😅

I've made the changes and added a few tests as appropriate. I found a few test files I found redundant (the submit ones for the 2 api types as they test the same thing and aren't specific to those tests) so I've removed them and put what they are testing in a more appropriate place.

VikaCep reviewed (approved):
LGTM! 🚀
